week3/ngrams/05_final_report.Rmd

```diff
@@ -145,7 +187,17 @@ For each year term, find the year where its proportion of mentions peaks (hits i
 Now, for each year term, find the minimum number of years it takes for the proportion of mentions to decline from its peak value to half its peak value. Store this in an intermediate data frame.
 
 ```{r compute-half-lifes}
-
+half_life_df <- inner_join(ngram_year_counts_and_totals, proportion_of_mention_peaks, by = "term") %>%
+  filter(year >= 1850 & year <= 2012) %>%
+  group_by(term) %>%
+  filter(proportion_of_mentions <= (proportion_of_mentions_highest * 0.5)) %>%
+  mutate(half_life = year - peak_year) %>%
+  filter(half_life > 0) %>% 

```

Comment:
This is a good, interesting solution that's different from how I'd do it. I'd do it by filtering (in the second filter) years after the peak year. But this is a perfectly cromulent way to do it too.